### ISSUE TITLE
GithubException: 500 None when using get_issues

### ISSUE THREAD

**Issue by fabianbaier** [PyGithub/PyGithub#912]
Hi,

When using ```
repo = g.get_repo("some/repo")
repo.get_issues(sort='created', direction='desc', state='closed')``` it fails after a few seconds with:
```
---------------------------------------------------------------------------
GithubException                           Traceback (most recent call last)
<timed exec> in <module>()

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in __iter__(self)
     62             yield element
     63         while self._couldGrow():
---> 64             newElements = self._grow()
     65             for element in newElements:
     66                 yield element

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _grow(self)
     74 
     75     def _grow(self):
---> 76         newElements = self._fetchNextPage()
     77         self.__elements += newElements
     78         return newElements

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _fetchNextPage(self)
    194             self.__nextUrl,
    195             parameters=self.__nextParams,
--> 196             headers=self.__headers
    197         )
    198         data = data if data else []

/opt/conda/lib/python3.6/site-packages/github/Requester.py in requestJsonAndCheck(self, verb, url, parameters, headers, input)
    260 
    261     def requestJsonAndCheck(self, verb, url, parameters=None, headers=None, input=None):
--> 262         return self.__check(*self.requestJson(verb, url, parameters, headers, input, self.__customConnection(url)))
    263 
    264     def requestMultipartAndCheck(self, verb, url, parameters=None, headers=None, input=None):

/opt/conda/lib/python3.6/site-packages/github/Requester.py in __check(self, status, responseHeaders, output)
    271         output = self.__structuredFromJson(output)
    272         if status >= 400:
--> 273             raise self.__createException(status, responseHeaders, output)
    274         return responseHeaders, output
    275 

GithubException: 500 None
```
Any way to work around this?

Interestingly when using `repo.get_issues(sort='created', direction='desc', state="open")` it works fine.

Any pointing in the right direction appreciated.

**Comment by sfdye** [comment]
I think this relates to #911

**Comment by singh811** [comment]
I tried the same thing on my end, but didn't get any error. I don't have much deeper knowledge of how Github api works, but can you please look out for the Silly Mistakes in your code which might be causing this. 
And if there is no such case then you can try the below given code in order to debug what may be causing the error.
```
import github

login = 'REDACTED' # replace it with your username
password = 'REDACTED' # replace it with your password

github.enable_console_debug_logging()

g = github.Github(login, password)
repo = g.get_repo('someuser/some-repo')

# This is just a seperator to seperate requests logs.
print("========================================================================")

issues = repo.get_issues(sort='created', direction='desc', state='closed')

for issue in issues:
    print issue.id
```

In the logs you can see which request is breaking and then we can investigate the issue further.

**Comment by fabianbaier** [comment]
Thanks @singh811 for your response! The repo I am testing it with is `kubernetes/kubernetes` and the more verbose output with `Python 3.6` looks like: 

```
GET https://api.github.com/repositories/20580498/issues?state=closed&sort=created&direction=desc&page=2 {'Authorization': 'token (oauth token removed)', 'User-Agent': 'PyGithub/Python'} None ==> 500 {'date': 'Mon, 01 Oct 2018 19:52:03 GMT', 'content-type': 'application/json; charset=utf-8', 'content-length': '0', 'server': 'GitHub.com', 'status': '500 Internal Server Error', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '3960', 'x-ratelimit-reset': '1538424013', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': 'repo', 'x-github-media-type': 'github.v3; format=json', 'access-control-expose-headers': 'ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '1; mode=block', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'x-runtime-rack': '6.739544', 'x-github-request-id': 'DGFD:06B5:266F949:3127690:5BBA7ADD'} 
---------------------------------------------------------------------------
GithubException                           Traceback (most recent call last)
<timed exec> in <module>()

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in __iter__(self)
     62             yield element
     63         while self._couldGrow():
---> 64             newElements = self._grow()
     65             for element in newElements:
     66                 yield element

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _grow(self)
     74 
     75     def _grow(self):
---> 76         newElements = self._fetchNextPage()
     77         self.__elements += newElements
     78         return newElements

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _fetchNextPage(self)
    194             self.__nextUrl,
    195             parameters=self.__nextParams,
--> 196             headers=self.__headers
    197         )
    198         data = data if data else []

/opt/conda/lib/python3.6/site-packages/github/Requester.py in requestJsonAndCheck(self, verb, url, parameters, headers, input)
    260 
    261     def requestJsonAndCheck(self, verb, url, parameters=None, headers=None, input=None):
--> 262         return self.__check(*self.requestJson(verb, url, parameters, headers, input, self.__customConnection(url)))
    263 
    264     def requestMultipartAndCheck(self, verb, url, parameters=None, headers=None, input=None):

/opt/conda/lib/python3.6/site-packages/github/Requester.py in __check(self, status, responseHeaders, output)
    271         output = self.__structuredFromJson(output)
    272         if status >= 400:
--> 273             raise self.__createException(status, responseHeaders, output)
    274         return responseHeaders, output
    275 

GithubException: 500 None
```

Hope that helps somehow for troubleshooting. I tried same code on `tensorflow/tensorflow` and it works just fine.

**Comment by fabianbaier** [comment]
the output is identical with your script. I had to modify your script to work in Python 3.6 (yours is in Python 2.x) and used token authentication instead of username/password.

```
import github

github.enable_console_debug_logging()

# using personalized Github access token to read public repos
g = github.Github("mytoken")
# declare which repo to look at
repo = g.get_repo("kubernetes/kubernetes")

# This is just a seperator to seperate requests logs.
print("========================================================================")

issues = repo.get_issues(sort='created', direction='desc', state='closed')

for issue in issues:
    print(issue.id)
```

Output (same problem):
```
GET https://api.github.com/repos/kubernetes/kubernetes {'Authorization': 'token (oauth token removed)', 'User-Agent': 'PyGithub/Python'} None ==> 200 {'date': 'Tue, 02 Oct 2018 17:19:51 GMT', 'content-type': 'application/json; charset=utf-8', 'transfer-encoding': 'chunked', 'server': 'GitHub.com', 'status': '200 OK', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4999', 'x-ratelimit-reset': '1538504391', 'cache-control': 'private, max-age=60, s-maxage=60', 'vary': 'Accept, Authorization, Cookie, X-GitHub-OTP, Accept-Encoding', 'etag': 'W/"70c897ac9639a7ffea2ded9983104771"', 'last-modified': 'Tue, 02 Oct 2018 17:11:51 GMT', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': 'repo', 'x-github-media-type': 'github.v3; format=json', 'access-control-expose-headers': 'ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '1; mode=block', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'x-runtime-rack': '0.081719', 'content-encoding': 'gzip', 'x-github-request-id': 'CC09:06B3:1242B9D:17234A7:5BB3A8B7'} {"id":20580498,"node_id":"MDEwOlJlcG9zaXRvcnkyMDU4MDQ5OA==","name":"kubernetes","full_name":"kubernetes/kubernetes","private":false,"owner":{"login":"kubernetes","id":13629408,"node_id":"MDEyOk9yZ2FuaXphdGlvbjEzNjI5NDA4","avatar_url":"https://avatars2.githubusercontent.com/u/13629408?v=4","gravatar_id":"","url":"https://api.github.com/users/kubernetes","html_url":"https://github.com/kubernetes","followers_url":"https://api.github.com/users/kubernetes/followers","following_url":"https://api.github.com/users/kubernetes/following{/other_user}","gists_url":"https://api.github.com/users/kubernetes/gists{/gist_id}","starred_url":"https://api.github.com/users/kubernetes/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/kubernetes/subscriptions","organizations_url":"https://api.github.com/users/kubernetes/orgs","repos_url":"https://api.github.com/users/kubernetes/repos","events_url":"https://api.github.com/users/kubernetes/events{/privacy}","received_events_url":"https://api.github.com/users/kubernetes/received_events","type":"Organization","site_admin":false},"html_url":"https://github.com/kubernetes/kubernetes","description":"Production-Grade Container Scheduling and Management","fork":false,"url":"https://api.github.com/repos/kubernetes/kubernetes","forks_url":"https://api.github.com/repos/kubernetes/kubernetes/forks","keys_url":"https://api.github.com/repos/kubernetes/kubernetes/keys{/key_id}","collaborators_url":"https://api.github.com/repos/kubernetes/kubernetes/collaborators{/collaborator}","teams_url":"https://api.github.com/repos/kubernetes/kubernetes/teams","hooks_url":"https://api.github.com/repos/kubernetes/kubernetes/hooks","issue_events_url":"https://api.github.com/repos/kubernetes/kubernetes/issues/events{/number}","events_url":"https://api.github.com/repos/kubernetes/kubernetes/events","assignees_url":"https://api.github.com/repos/kubernetes/kubernetes/assignees{/user}","branches_url":"https://api.github.com/repos/kubernetes/kubernetes/branches{/branch}","tags_url":"https://api.github.com/repos/kubernetes/kubernetes/tags","blobs_url":"https://api.github.com/repos/kubernetes/kubernetes/git/blobs{/sha}","git_tags_url":"https://api.github.com/repos/kubernetes/kubernetes/git/tags{/sha}","git_refs_url":"https://api.github.com/repos/kubernetes/kubernetes/git/refs{/sha}","trees_url":"https://api.github.com/repos/kubernetes/kubernetes/git/trees{/sha}","statuses_url":"https://api.github.com/repos/kubernetes/kubernetes/statuses/{sha}","languages_url":"https://api.github.com/repos/kubernetes/kubernetes/languages","stargazers_url":"https://api.github.com/repos/kubernetes/kubernetes/stargazers","contributors_url":"https://api.github.com/repos/kubernetes/kubernetes/contributors","subscribers_url":"https://api.github.com/repos/kubernetes/kubernetes/subscribers","subscription_url":"https://api.github.com/repos/kubernetes/kubernetes/subscription","commits_url":"https://api.github.com/repos/kubernetes/kubernetes/commits{/sha}","git_commits_url":"https://api.github.com/repos/kubernetes/kubernetes/git/commits{/sha}","comments_url":"https://api.github.com/repos/kubernetes/kubernetes/comments{/number}","issue_comment_url":"https://api.github.com/repos/kubernetes/kubernetes/issues/comments{/number}","contents_url":"https://api.github.com/repos/kubernetes/kubernetes/contents/{+path}","compare_url":"https://api.github.com/repos/kubernetes/kubernetes/compare/{base}...{head}","merges_url":"https://api.github.com/repos/kubernetes/kubernetes/merges","archive_url":"https://api.github.com/repos/kubernetes/kubernetes/{archive_format}{/ref}","downloads_url":"https://api.github.com/repos/kubernetes/kubernetes/downloads","issues_url":"https://api.github.com/repos/kubernetes/kubernetes/issues{/number}","pulls_url":"https://api.github.com/repos/kubernetes/kubernetes/pulls{/number}","milestones_url":"https://api.github.com/repos/kubernetes/kubernetes/milestones{/number}","notifications_url":"https://api.github.com/repos/kubernetes/kubernetes/notifications{?since,all,participating}","labels_url":"https://api.github.com/repos/kubernetes/kubernetes/labels{/name}","releases_url":"https://api.github.com/repos/kubernetes/kubernetes/releases{/id}","deployments_url":"https://api.github.com/repos/kubernetes/kubernetes/deployments","created_at":"2014-06-06T22:56:04Z","updated_at":"2018-10-02T17:11:51Z","pushed_at":"2018-10-02T17:11:50Z","git_url":"git://github.com/kubernetes/kubernetes.git","ssh_url":"git@github.com:kubernetes/kubernetes.git","clone_url":"https://github.com/kubernetes/kubernetes.git","svn_url":"https://github.com/kubernetes/kubernetes","homepage":"https://kubernetes.io","size":811870,"stargazers_count":42115,"watchers_count":42115,"language":"Go","has_issues":true,"has_projects":true,"has_downloads":true,"has_wiki":false,"has_pages":false,"forks_count":14605,"mirror_url":null,"archived":false,"open_issues_count":3181,"license":{"key":"apache-2.0","name":"Apache License 2.0","spdx_id":"Apache-2.0","url":"https://api.github.com/licenses/apache-2.0","node_id":"MDc6TGljZW5zZTI="},"forks":14605,"open_issues":3181,"watchers":42115,"default_branch":"master","permissions":{"admin":false,"push":false,"pull":true},"organization":{"login":"kubernetes","id":13629408,"node_id":"MDEyOk9yZ2FuaXphdGlvbjEzNjI5NDA4","avatar_url":"https://avatars2.githubusercontent.com/u/13629408?v=4","gravatar_id":"","url":"https://api.github.com/users/kubernetes","html_url":"https://github.com/kubernetes","followers_url":"https://api.github.com/users/kubernetes/followers","following_url":"https://api.github.com/users/kubernetes/following{/other_user}","gists_url":"https://api.github.com/users/kubernetes/gists{/gist_id}","starred_url":"https://api.github.com/users/kubernetes/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/kubernetes/subscriptions","organizations_url":"https://api.github.com/users/kubernetes/orgs","repos_url":"https://api.github.com/users/kubernetes/repos","events_url":"https://api.github.com/users/kubernetes/events{/privacy}","received_events_url":"https://api.github.com/users/kubernetes/received_events","type":"Organization","site_admin":false},"network_count":14605,"subscribers_count":2746}
========================================================================
GET https://api.github.com/repos/kubernetes/kubernetes/issues?state=closed&sort=created&direction=desc {'Authorization': 'token (oauth token removed)', 'User-Agent': 'PyGithub/Python'} None ==> 500 {'date': 'Tue, 02 Oct 2018 17:20:00 GMT', 'content-type': 'application/json; charset=utf-8', 'content-length': '0', 'server': 'GitHub.com', 'status': '500 Internal Server Error', 'x-ratelimit-limit': '5000', 'x-ratelimit-remaining': '4998', 'x-ratelimit-reset': '1538504391', 'x-oauth-scopes': 'repo', 'x-accepted-oauth-scopes': 'repo', 'x-github-media-type': 'github.v3; format=json', 'access-control-expose-headers': 'ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', 'access-control-allow-origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubdomains; preload', 'x-frame-options': 'deny', 'x-content-type-options': 'nosniff', 'x-xss-protection': '1; mode=block', 'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin', 'content-security-policy': "default-src 'none'", 'x-runtime-rack': '9.309939', 'x-github-request-id': 'CC09:06B3:1242BA6:17234B3:5BB3A8B7'} 
---------------------------------------------------------------------------
GithubException                           Traceback (most recent call last)
<ipython-input-6-dd0c03de4039> in <module>()
     13 issues = repo.get_issues(sort='created', direction='desc', state='closed')
     14 
---> 15 for issue in issues:
     16     print(issue.id)

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in __iter__(self)
     62             yield element
     63         while self._couldGrow():
---> 64             newElements = self._grow()
     65             for element in newElements:
     66                 yield element

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _grow(self)
     74 
     75     def _grow(self):
---> 76         newElements = self._fetchNextPage()
     77         self.__elements += newElements
     78         return newElements

/opt/conda/lib/python3.6/site-packages/github/PaginatedList.py in _fetchNextPage(self)
    194             self.__nextUrl,
    195             parameters=self.__nextParams,
--> 196             headers=self.__headers
    197         )
    198         data = data if data else []

/opt/conda/lib/python3.6/site-packages/github/Requester.py in requestJsonAndCheck(self, verb, url, parameters, headers, input)
    260 
    261     def requestJsonAndCheck(self, verb, url, parameters=None, headers=None, input=None):
--> 262         return self.__check(*self.requestJson(verb, url, parameters, headers, input, self.__customConnection(url)))
    263 
    264     def requestMultipartAndCheck(self, verb, url, parameters=None, headers=None, input=None):

/opt/conda/lib/python3.6/site-packages/github/Requester.py in __check(self, status, responseHeaders, output)
    271         output = self.__structuredFromJson(output)
    272         if status >= 400:
--> 273             raise self.__createException(status, responseHeaders, output)
    274         return responseHeaders, output
    275 

GithubException: 500 None
```

As you can see same output as the above in my last post. Is there a problem with the request itself and its endpoint?

I am asking because 
```
nobody@621b826296ea:~/work$ curl -s -v https://api.github.com/repos/kubernetes/kubernetes/issues?state=closed&sort=created&direction=desc
*   Trying 192.30.255.116...
* TCP_NODELAY set
* Connected to api.github.com (192.30.255.116) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /opt/conda/ssl/cacert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Jun 19 00:00:00 2018 GMT
*  expire date: Jul 10 12:00:00 2019 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*  SSL certificate verify ok.
> GET /repos/kubernetes/kubernetes/issues?state=closed HTTP/1.1
> Host: api.github.com
> User-Agent: curl/7.61.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< Date: Tue, 02 Oct 2018 17:34:19 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 0
< Server: GitHub.com
< Status: 500 Internal Server Error
< X-RateLimit-Limit: 60
< X-RateLimit-Remaining: 44
< X-RateLimit-Reset: 1538504749
< X-GitHub-Media-Type: github.v3; format=json
< Access-Control-Expose-Headers: ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
< Access-Control-Allow-Origin: *
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< X-Frame-Options: deny
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
< Content-Security-Policy: default-src 'none'
< X-Runtime-rack: 5.830364
< X-GitHub-Request-Id: CC70:06B4:255217F:301BA4A:5BB3AC15
<
* Connection #0 to host api.github.com left intact
```

Why would api.github.com return `500` when I am authenticated and not rate-limited ?

I can now reproduce `500` pretty steady, its almost like every other request ends up in a `500`.
